cluster/engine.go

```diff
@@ -347,6 +347,15 @@ func (e *Engine) TotalCpus() int64 {
 	return e.Cpus + (e.Cpus * e.overcommitRatio / 100)
 }
 
+// NodeLabels returns the assigned and standard labels
+func (e *Engine) NodeLabels() string {
+	labels := []string{}
```

Comment:
I don't want to be annoying :D, but 
```go
labels = make([]string, 0, len(e.Labels)
```
It will accelerate this method on 25-30% depending of the amount of keys.